utils/mountd/cache.c

```diff
@@ -334,6 +338,15 @@ static int uuid_by_path(char *path, int type, size_t uuidlen, char *uuid)
 	const char *val;
 	int rc;
 
+	/* Delphix -- Use cached fsid value if available */
```

Comment:
It looks we should only get the cached fsid if (type == 0) but go through the old code path. As is, we get the cached fsid (if one exists) then return 0 when type != 0.

That said, I'm not sure if it would makes the difference given uuid_by_path() invocations use type=0 or starts at 0.

Reply:
good point.  I'll wrap the checking for a cached value in a `if (type == 0)` block.